src/document.c

```diff
@@ -135,6 +135,12 @@ RSAddDocumentCtx *NewAddDocumentCtx(IndexSpec *sp, Document *b, const char **err
     aCtx->fwIdx = NewForwardIndex(&aCtx->doc, sp->flags);
   }
 
+  if (sp->smap){
+    aCtx->fwIdx->smap = SynonymMap_GetReadOnlyCopy(sp->smap);
+  }else{
```

Comment:
spacing again here.. i use vscode for formatting and you should too; makes things easier. do you have a preferred editor?

Reply:
Just installed vscode, I am learning how to use it.